src/fastmcp/mcp_config.py

```diff
@@ -162,85 +193,62 @@ def to_transport(self) -> StreamableHttpTransport | SSETransport:
             )
 
 
+class TransformingRemoteMCPServer(WrappedMCPServerMixin, RemoteMCPServer):
+    """A Remote server with tool transforms."""
+
+
+MCPServerTypes = (
+    StdioMCPServer
+    | TransformingStdioMCPServer
+    | RemoteMCPServer
+    | TransformingRemoteMCPServer
+)
+
+McpServersType = dict[str, MCPServerTypes]
```

Comment:
Weak opinion that having both `MCPServerTypes` and `McpServersType` in this file is going to cause more difficult-to-debug typing issues than it prevents. If possible I'd recommend sticking with `dict[str, MCPServerTypes]` as its much clearer and will save users from the risk of accidentally misspelling the import if they actually need to reference this type in another file.